### PR TITLE
Enable us to handle version control via Plugin_Register files

### PR DESCRIPTION
**Pull request by bordoni**
When reviewing please make sure to disable white-space.

### CODE REVIEW

File: src/Command.php [hunk 1]
```diff
@@ -118,23 +118,25 @@ protected function configure() {
 	}
 
 	public function get_plugin( $search ) {
```
Comment (Camwyn) [inline src/Command.php:120]: Docblock

File: src/Command.php [hunk 2]
```diff
@@ -146,6 +148,64 @@ function ( $plugin ) use ( $search ) {
...
 	}
 
+	/**
+	 * Given a Plugin Object, we determine what is the version storage file path.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed $plugin An object of the plugin, based on tut.json structure.
+	 *
+	 * @return null|string
```
Comment (Camwyn) [inline src/Command.php:158]: Why null and not an empty string?

File: src/Command.php [hunk 3]
```diff
@@ -146,6 +148,64 @@ function ( $plugin ) use ( $search ) {
...
+	 * Given a Plugin array data, apply the defaults so we dont have to check before using them.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $plugin
+	 *
+	 * @return array
+	 */
+	protected function apply_plugin_default_args( array $plugin ): array {
+		$default_args = [
```
Comment (Camwyn) [inline src/Command.php:192]: Again, most of these are strings, why "convert" to null if not set?

**Review by Camwyn — APPROVED** [review]

File: src/Command.php (hunk 2 above)
Comment (bordoni) [reply src/Command.php:158]: I wanted to leave room for the diff in value. Like empty would be something else in the future

**Comment by borkweb** [comment]
Here's a commit that supports putting the version in `Plugin_Register` in a bit more simple way: https://github.com/the-events-calendar/tut/commit/c3968dfcfaa1b5cbea962a7ff83c036de1175240

**Review by borkweb — CHANGES_REQUESTED** [review]
We shouldn't need all of these changes, luckily! https://github.com/the-events-calendar/tut/commit/c3968dfcfaa1b5cbea962a7ff83c036de1175240

**Pull request closed without merging**